### DISCUSSION TITLE
Dynamic Route overriding public folder

### DISCUSSION THREAD

**Discussion by AnuragRamdasan** [vercel/next.js#19451] [Help] [answered]
I have created a Dynamic Route Component called [base].js in my src. I needed to do it this way because I am porting a project to next js so I needed to overwrite the path at the root.

However this seems to be bloging my access to `public` folder. A lookup for images in `public` routes me to the `[base].js` and ends up in a 404.

Is there a best practice around this? Or even a way for my fallback to somehow route to the public folder?

**Comment by Tomburgs** [comment, marked as answer]
Could you share your project's structure?
By default next should check for the file in your public folder and only then match a route.

What do you mean by an image lookup routes you to `[base].js` and results into 404, why does your `[base].js` return a 404?
You mention that the `[base].js` is in your `src` directory, but it should be under the `pages` directory, could this be the issue?

**Reply by AnuragRamdasan** [reply]
Hi Thanks for the reply!

here's my folder structure excluding node_modules

── README.md
├── cat.json
├── firebase.json
├── firebaseFunctions.js
├── package-lock.json
├── package.json
├── public
│   ├── favicon.ico
│   ├── img
├── src
│   ├── components
│   ├── content
│   ├── next.config.js
│   ├── pages
│   ├── styles
│   └── utils
└── yarn.lock

[base].js lives inside src/pages

This means [base].js is the catchall route page. However due to this, a /favicon.ico will also end up being passed to [base].js instead of public/favicon.ico, or atleast that is happening in my case.

**Reply by Tomburgs** [reply]
The problem seems to be with your scope, from the fact that your next.config.js is under src directory I take it that you are running the next commands from there?
Otherwise, I don't see how your config would get read.

If that's the case then that's also where the public directory should be, however I suggest that you move your config to root and execute the next commands from there instead.

**Reply by AnuragRamdasan** [reply]
Ah got it. This is super helpful. I didnt know thats how it had to be structured.

This resolves my issue. Thanks a lot!
